src/Ledger/Gov.lagda

```diff
@@ -29,6 +33,41 @@ record GovEnv : Set where
         epoch       : Epoch
         pparams     : PParams
         enactState  : EnactState
+
+-- getAidPairs returns a list of (aid , aidₚ), where aid is an action id and
+-- aidₚ is the id of the previous action.
+getAidPairs : GovState → List (GovActionID × GovActionID)
+getAidPairs aid×stateList =
+  mapMaybe (λ (aid , aState) → (aid ,_) <$> getHash (prevAction aState)) $ aid×stateList
+  where open GovActionState
+
+connects : List (GovActionID × GovActionID) → GovActionID × GovActionID → Set
+connects [] (aid , aidₚ) = aid ≡ aidₚ
+connects ((a , aₚ) ∷ s) (aid , aidₚ) = connects s (a , aid) × aₚ ≡ aidₚ
+
+connects? : ∀ l aidPair → Dec (connects l aidPair)
+connects? [] (aid , aidₚ) = aid ≟ aidₚ
+connects? ((a , aₚ) ∷ s) (aid , aidₚ) with connects? s (a , aid) | (aₚ ≟ aidₚ)
+...| yes p  | yes q  = yes (p , q)
+...| _      | no ¬q  = no λ (_ , q) → ¬q q
+...| no ¬p  | _      = no λ (p , _) → ¬p p
+
+enactable : EnactState → GovActionID × GovActionState → List (GovActionID × GovActionID) → Set
+enactable eState (aid , record { action = action }) aidPairs =
```

Comment:
There are two issues with this definition:
- I made a mistake in the original definition, we need to make sure that (the ID inside) `getHash (GovActionState.prevAction as)` connects to `aidₚ`, instead of `aid`. `enactable` should not depend on `aid` at all.
- Using `subpermutations` makes the definition of `enactable?` easier, but it makes any other proofs about it way harder, since they now need to unfold the definition of `subpermutations`. Also, this definition will show up in the PDF, so the reader would need to know the definition of `subpermutations`. It would be much better to revert this to 'there exists a subset such that ...'.

Reply:
I'm okay with the first point, and all the comments above.  I'll fix those.

Re. the second point, [we did discuss](https://input-output-rnd.slack.com/archives/D04JE60R5T3/p1701949721662799) using the collection of all permutations of all sublists.  If I revert to a set-based approach, the proofs will be much harder, and we will still need a function for getting all subsets of a given set and testing equality of each of these sets with each candidate.  Later proofs will still need to unfold the definitions of these functions.

I would like to fix the issues mentioned in the first point and above comments and then I would suggest we get this list-based solution merged, even if it's not the ideal solution.  It works, and, from my experience over the past week or so, a set-based solution will be much harder to get working and to reason about.

Reply:
Okay, I'll spend a few more hours on the set-based proof and see if I can get it working.  If I get stuck, I'll ping you (@WhatisRT) on slack.  Thanks!